pkg/docker/config/config.go

```diff
@@ -222,14 +226,13 @@ func getAuthFilePaths(sys *types.SystemContext, homeDir string) []authPath {
 		if xdgCfgHome == "" {
 			xdgCfgHome = filepath.Join(homeDir, ".config")
 		}
-		paths = append(paths, authPath{path: filepath.Join(xdgCfgHome, xdgConfigHomePath), legacyFormat: false})
+		paths = append(paths, newAuthPathDefault(filepath.Join(xdgCfgHome, xdgConfigHomePath)))
 		if dockerConfig := os.Getenv("DOCKER_CONFIG"); dockerConfig != "" {
-			paths = append(paths,
-				authPath{path: filepath.Join(dockerConfig, "config.json"), legacyFormat: false},
-			)
+			path := newAuthPathDefault(filepath.Join(dockerConfig, "config.json"))
+			paths = append(paths, path)
```

Comment:
Inline the `path` value into its single user? All the surrounding code inlines `append(paths, newAuthPathDefault(…))` as well, this one being different invites worrying what’s special about it.